src/Hal/Serializer/ItemNormalizer.php

```diff
@@ -99,8 +102,10 @@ protected function getAttributes($object, $format = null, array $context)
      */
     private function getComponents($object, string $format = null, array $context)
     {
-        if (false !== $context['cache_key'] && isset($this->componentsCache[$context['cache_key']])) {
-            return $this->componentsCache[$context['cache_key']];
+        $cacheKey = \get_class($object).'-'.$context['cache_key'];
```

Comment:
We know the object class in the `normalize` method right? Could we pass the information to this function to avoid calling `get_class`? Might be a micro-optimization though. (same below)

Reply:
`\get_class` is transformed in opcode, it's almost costless.

Reply:
For info I've copied the method used in core symfony, which is to add the class here